### ISSUE TITLE
Allow digits in the local part of prefixed names

### ISSUE THREAD

**Issue by mightymax** [rdfjs/N3.js#469]
When creating Turtle files with using prefixes where the local part of prefixed names starts with a number, the N3 Writer does not use the declared prefix.

Minimal working example:
```js
import { Writer, DataFactory } from 'n3';
const { quad, namedNode, literal } = DataFactory;
const writer = new Writer({ prefixes: { c: 'http://example.org/' } })
writer.addQuad(quad(
  namedNode('http://example.org/1'),
  namedNode('http://example.org/2'),
  literal('Testing localnames with numbers')
));
writer.end((_, result) => console.log(result));
```

**Expected** output:
```ttl
@prefix c: <http://example.org/>.
c:1 c:2 "Tetsing localnames with numbers".
```

**Actual** output:
```tel
@prefix c: <http://example.org/>.
<http://example.org/1> <http://example.org/2> "Testing local names with numbers".
```

The [Turtle standard, section on prefixed names](https://www.w3.org/TR/turtle/#prefixed-name) notes that leading digits are allowed as local part of prefixed names in the superset of allowed XML QNames.

Here is the diff that solves this issue:

```diff
diff --git a/node_modules/n3/lib/N3Writer.js b/node_modules/n3/lib/N3Writer.js
index 58fb556..32044fa 100644
--- a/node_modules/n3/lib/N3Writer.js
+++ b/node_modules/n3/lib/N3Writer.js
@@ -284,7 +284,7 @@ class N3Writer {
         prefixList += (prefixList ? '|' : '') + this._prefixIRIs[prefixIRI];
       }
       IRIlist = escapeRegex(IRIlist, /[\]\/\(\)\*\+\?\.\\\$]/g, '\\$&');
-      this._prefixRegex = new RegExp(`^(?:${prefixList})[^\/]*$|` + `^(${IRIlist})([_a-zA-Z][\\-_a-zA-Z0-9]*)$`);
+      this._prefixRegex = new RegExp(`^(?:${prefixList})[^\/]*$|` + `^(${IRIlist})([_a-zA-Z0-9][\\-_a-zA-Z0-9]*)$`);
     }
     // End a prefix block with a newline
     this._write(hasPrefixes ? '\n' : '', done);
```

**Comment by jeswr** [comment]
Thanks @mightymax, this change seems reasonable.

Could you open a PR for this that also contains a regression test?

**Comment by mightymax** [comment]
Done!

**Comment by jeswr** [comment]
In future when using GitHub I'd suggest writing "closes #469" in your PR description. This links the issue and triggers the issue to close automatically when the PR is merged.

This means, e.g., issues stay open of PRs go stale / are closed without merging.
